### PR TITLE
PM-5019 Migrate Adjust Payment component

### PR DESCRIPTION
**Pull request by KiruthigaManivannan**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other
```

## Objective

Migrate the Adjust Payment component to use the component library.

## Code changes

- **adjust-payment-dialog.component.html:** Updated Form, Cards to use the equivalent components from the Component Library. Cards are updated to Dialog component from component library.
- **adjust-payment-dialog.component.ts:** Added FormGroup. Used Dialog service to create a dialog open method. Dialog data interface and dialog result const is created.
- **payment-method.component.html:** Removed the app-adjust-payment selector and properties bound to it.
- **payment-method.component.ts:** Adjust payment dialog is opened during the onclick of changePayment button. Final result of the dialog is captured and loaded again

## Screenshots
-**Click on Change Payment Method button**
Before:
image
After:
image

-**Select credit card option and fill in details
Before:
image
image

After:
image
image

-**Select paypal option and fill in details
Before:
image
After:
image

### CODE REVIEW

File: apps/web/src/app/billing/shared/payment-method.component.html [hunk 1]
```diff
@@ -15,10 +15,9 @@
 
 <bit-container>
   <div class="tabbed-header" *ngIf="!organizationId">
-    <!-- TODO: Organization and individual should use different "page" components -->
+    <!--TODO: Organization and individual should use different "page" components -->
     <h1>{{ "paymentMethod" | i18n }}</h1>
   </div>
-
   <ng-container *ngIf="!firstLoaded && loading">
```
Comment (willmartian) [inline apps/web/src/app/billing/shared/payment-method.component.html:21]: These diffs are unrelated to the PR

File: apps/web/src/app/billing/shared/adjust-payment-dialog.component.ts [hunk 2]
```diff
@@ -66,14 +81,15 @@ export class AdjustPaymentComponent {
...
     } catch (e) {
       this.logService.error(e);
+      throw e;
     }
-  }
+  };
 
   cancel() {
-    this.onCanceled.emit();
+    this.dialogRef.close(AdjustPaymentDialogResult.Cancelled);
```
Comment (willmartian) [inline apps/web/src/app/billing/shared/adjust-payment-dialog.component.ts:92]: We never call this method anywhere. Either call it with the cancel button, or pass the close value to `bitDialogClose`

**Review by willmartian — CHANGES_REQUESTED** [review]

**Review by sukhleenb — APPROVED** [review]

File: apps/web/src/app/billing/shared/payment-method.component.html (hunk 1 above)
Comment (KiruthigaManivannan) [reply apps/web/src/app/billing/shared/payment-method.component.html:21]: Sorry @willmartian For testing purpose i added a line there. Only after I committed the i understood that i need to remove the test code. So After a commit i removed the code thats why it is showing like line is changed.

Comment (willmartian) [reply apps/web/src/app/billing/shared/payment-method.component.html:21]: No worries. Could you revert the change?

**Review by willmartian — CHANGES_REQUESTED** [review]
Looks good! Just one small change still needed here: https://github.com/bitwarden/clients/pull/8383/files#r1532702398

**Review by willmartian — APPROVED** [review]

**Review by cturnbull-bitwarden — APPROVED** [review]

**Pull request merged**